src/Integration.Vsix/Suppression/SuppressionManager.cs

```diff
@@ -57,13 +56,21 @@ public SuppressionManager(IServiceProvider serviceProvider, ITimerFactory timerF
 
         private void RefreshSuppresionHandling()
         {
-            if (activeSolutionBoundTracker.IsActiveSolutionBound)
+            // This method can be called on the UI thread so unhandled exceptions will crash VS
+            try
             {
-                SetupSuppressionHandling();
+                if (activeSolutionBoundTracker.IsActiveSolutionBound)
+                {
+                    SetupSuppressionHandling();
+                }
+                else
+                {
+                    CleanupSuppressionHandling();
+                }
             }
-            else
+            catch(Exception)
             {
-                CleanupSuppressionHandling();
+                // TODO: log exceptions to the output window
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)